manager/state/raft/raft.go

```diff
@@ -461,14 +459,13 @@ func (n *Node) Run(ctx context.Context) error {
 			// If we are the only registered member after
 			// restoring from the state, campaign to be the
 			// leader.
-			if !n.restored {
-				// Node ID should be in the progress list to Campaign
-				if len(n.cluster.Members()) <= 1 {
+			if n.campaignWhenAble && n.addedSelf {
+				if len(n.cluster.Members()) == 1 {
```

Comment:
Hm. For some reason I thought there were situations where we could call `n.cluster.AddMember()` without actually applying a conf change in raft, but examining the code, I can't see any.

Updated. I'll update the 1.12 version as well.